src/collections/skip_list/mod.rs

```diff
@@ -0,0 +1,449 @@
+        write!(f, "{:?}: {:?}", self.key, self.value)
+    }
+}
+
+struct SkipNode<K, V> {
+    entry: Entry<K, V>,
+    next_by_height: SkipTrack<K, V>,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rand::RngCore;
+    use std::cmp::Ordering;
+    use std::collections::BTreeMap;
+    use std::time::Instant;
+
+    fn create_testee() -> SkipList<String, i32> {
+        let mut list = SkipList::new();
+        let test_case = [("c", 4i32), ("e", 5i32), ("a", 3), ("a", 8)];
+        test_case
+            .iter()
+            .for_each(|&(k, v)| list.insert(k.to_owned(), v));
+        list
+    }
+
+    #[test]
+    fn insert() {
+        let list = create_testee();
+        let res: Vec<(String, i32)> = list.into_iter().collect();
+        assert_eq!(
+            vec![
+                ("a".to_owned(), 8),
+                ("c".to_owned(), 4),
+                ("e".to_owned(), 5)
+            ],
+            res
+        );
+    }
+
+    #[test]
+    fn insert_get() {
+        let mut list = create_testee();
+        assert_eq!(list.get(&"c".to_owned()), Some(&4));
+        assert_eq!(list.get(&"e".to_owned()), Some(&5));
+        assert_eq!(list.get(&"x".to_owned()), None);
+        list.insert("x".to_owned(), 100);
+        assert_eq!(list.get(&"x".to_owned()), Some(&100));
+    }
+
+    #[test]
+    fn pop_and_into_iter() {
+        let mut list = SkipList::new();
+        let test_case = [("c", 4), ("e", 5), ("a", 3), ("a", 8)];
+        test_case.iter().for_each(|&(k, v)| list.insert(k, v));
+        assert_eq!(list.len(), 3);
+        assert_eq!(
+            list.pop_first().map(|x| Entry::into_tuple(x)),
+            Some(("a", 8))
+        );
+        assert_eq!(
+            list.pop_first().map(|x| Entry::into_tuple(x)),
+            Some(("c", 4))
+        );
+        assert_eq!(
+            list.pop_first().map(|x| Entry::into_tuple(x)),
+            Some(("e", 5))
+        );
+        assert_eq!(list.pop_first().map(|x| Entry::into_tuple(x)), None);
+        assert_eq!(list.pop_first().map(|x| Entry::into_tuple(x)), None);
+        assert_eq!(list.len(), 0);
+        test_case.iter().for_each(|&(k, v)| list.insert(k, v));
+        assert_eq!(list.len(), 3);
+        assert_eq!(
+            list.into_iter().collect::<Vec<(&str, i32)>>(),
+            vec![("a", 8), ("c", 4), ("e", 5)]
+        );
+    }
+
+    #[test]
+    fn test_fn_keys_values() {
+        let list = create_testee();
+        assert_eq!(
+            vec![&"a".to_owned(), &"c".to_owned(), &"e".to_owned()],
+            list.keys().collect::<Vec<&String>>()
+        );
+        assert_eq!(vec![&8, &4, &5], list.values().collect::<Vec<&i32>>());
+    }
+
+    #[test]
+    fn test_get_mut() {
+        let mut list = create_testee();
+        let mut ele = list.get_mut(&"a".to_owned()).unwrap();
+        *ele = 114514;
+        assert_eq!(*list.get_mut(&"a".to_owned()).unwrap(), 114514);
+    }
+
+    #[test]
```

Comment:
Thought this should be a [`#[bench]`](https://doc.rust-lang.org/nightly/unstable-book/library-features/test.html) test but not unit test.